content/posts/Docker Tags.md

```diff
@@ -0,0 +1,54 @@
+
+Instead of numbered version tag, you could specify `FROM python@sha256:7e6c00cc553fdce06c1bcfcbf34c73a0f3623a8fc9ce88c8fb91671f2cbc0dab` in your Dockerfile. This is the exact fingerprint of a specific image, and it will never change. This would ensure that nothing about your base image would ever change out from underneath you. This means you probably won't see as many unexpected build failures when some downstream packages. The tradeoff is you won’t be pulling in any newer versions of base images or packages that contain updates, like security patches.
+
+To sum is up: When using something like the `:latest` tag of a public Docker image you are basically always going to be running a patched version of that container, but the packages inside will constantly be changing. By using a specific `sha256` the image won’t change, meaning you won’t get any patches or updates without updating the tag in your Dockerfile.
+
+So which strategy should you use? Ultimately this should be a discussion you will want to have with anyone working on the codebase. Depending on a number of factors such as the number of people working on an application, the resources the team has available, the complexity of the application, and the language used, one tagging strategy may be more appropriate over another.
+
+If you are looking for a place to start out, try starting with using the minor stable release of the language you are using, such as `python:3.7`. Or in the case of ruby, try something like `ruby:2.6`. For Go, start with something like `golang:1.12`, (or try the `alpine` or `slim` versions if you want to use much smaller images). If you run into some unexpected behavior in testing or are constantly having build failures from dependencies changing out from underneath you, move to a more specific tag. You may end up deciding that for that particular Dockerfile you want to use the exact `sha256` for your base container. This is a fine option, just remember that it means you’ll also need to regularly review and update your base image manually in order to pull in any security patches.
+
+Hopefully this has been a useful look at Docker version tags, and the implications of using either a very generic or very specific tag when writing your Dockerfiles. You will find plenty of examples of applications out in the wild using all variations of the options we have discussed. And you will converse with folks who may have very strong opinions on the subject, one way or the other. We encourage you to experiment, understand the pros and cons of all options, and decide in collaboration with your teams what makes sense for any given application. And don’t be afraid to change course if one tagging strategy doesn’t seem to be working out.
```

Comment:
Never say 'Hopefully this has been useful'—it sounds like you think it hasn't. Point out that it's a live issue, and there's no one right answer that applies to everybody. Finally, please tie in to the book, with a link to the 'big day' post, and mention that we have an expanded discussion of this in Chapter XX.